soc/intel/atom/soc.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2010-2015, Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Board configuration macros for the ia32 platform
+ *
+ * This header file is used to specify and describe board-level aspects for
+ * the 'ia32' platform.
+ */
+
+#ifndef __SOC_H_
+#define __SOC_H_
+
+#include <zephyr/sys/util.h>
+
+#ifndef _ASMLANGUAGE
+#include <zephyr/device.h>
+#include <zephyr/random/random.h>
+#endif
```

Comment:
none of these seem to be required by this file, only toolchain.h for ARG_UNUSED?